x/staking/keeper/unbonding_test.go

```diff
@@ -0,0 +1,332 @@
+	delAddrs, valAddrs := createValAddrs(2)
+
+	type exists struct {
+		setRedelegation              bool
+		setRedelegationByUnbondingID bool
+	}
+
+	cases := []struct {
+		exists   exists
+		name     string
+		expected types.Redelegation
+	}{
+		{
+			name:   "existing 1",
+			exists: exists{true, true},
+			expected: types.NewRedelegation(
+				delAddrs[0],
+				valAddrs[0],
+				valAddrs[1],
+				0,
+				time.Unix(5, 0).UTC(),
+				sdk.NewInt(10),
+				math.LegacyNewDec(10),
+				0,
+			),
+		},
+		{
+			name:   "not existing 1",
+			exists: exists{false, true},
+			expected: types.NewRedelegation(
+				delAddrs[1],
+				valAddrs[0],
+				valAddrs[1],
+				0,
+				time.Unix(5, 0).UTC(),
+				sdk.NewInt(10),
+				math.LegacyNewDec(10),
+				0,
+			),
+		},
+		{
+			name:   "not existing 2",
+			exists: exists{false, false},
+			expected: types.NewRedelegation(
+				delAddrs[1],
+				valAddrs[1],
+				valAddrs[0],
+				0,
+				time.Unix(5, 0).UTC(),
+				sdk.NewInt(10),
+				math.LegacyNewDec(10),
+				0,
+			),
+		},
+	}
+
+	for i, tc := range cases {
+		s.Run(tc.name, func() {
+			if tc.exists.setRedelegation {
+				s.stakingKeeper.SetRedelegation(s.ctx, tc.expected)
+			}
+
+			if tc.exists.setRedelegationByUnbondingID {
+				s.stakingKeeper.SetRedelegationByUnbondingID(s.ctx, tc.expected, uint64(i))
+			}
+
+			red, found := s.stakingKeeper.GetRedelegationByUnbondingID(s.ctx, uint64(i))
+			if tc.exists.setRedelegation && tc.exists.setRedelegationByUnbondingID {
+				s.Require().True(found)
+				s.Require().Equal(tc.expected, red)
+			} else {
+				s.Require().False(found)
+			}
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestValidatorByUnbondingIDAccessors() {
+	_, valAddrs := createValAddrs(3)
+
+	type exists struct {
+		setValidator              bool
+		setValidatorByUnbondingID bool
+	}
+
+	cases := []struct {
+		exists    exists
+		name      string
+		validator types.Validator
+	}{
+		{
+			name:      "existing 1",
+			exists:    exists{true, true},
+			validator: testutil.NewValidator(s.T(), valAddrs[0], PKs[0]),
+		},
+		{
+			name:      "not existing 1",
+			exists:    exists{false, true},
+			validator: testutil.NewValidator(s.T(), valAddrs[1], PKs[1]),
+		},
+		{
+			name:      "not existing 2",
+			exists:    exists{false, false},
+			validator: testutil.NewValidator(s.T(), valAddrs[2], PKs[0]),
+		},
+	}
+
+	for i, tc := range cases {
+		s.Run(tc.name, func() {
+			if tc.exists.setValidator {
+				s.stakingKeeper.SetValidator(s.ctx, tc.validator)
+			}
+
+			if tc.exists.setValidatorByUnbondingID {
+				s.stakingKeeper.SetValidatorByUnbondingID(s.ctx, tc.validator, uint64(i))
+			}
+
+			val, found := s.stakingKeeper.GetValidatorByUnbondingID(s.ctx, uint64(i))
+			if tc.exists.setValidator && tc.exists.setValidatorByUnbondingID {
+				s.Require().True(found)
+				s.Require().Equal(tc.validator, val)
+			} else {
+				s.Require().False(found)
+			}
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestUnbondingCanComplete() {
```

Comment:
@julienrbrt I believe this is providing less coverage than the old tests were

Reply:
Yes, there are simple unit tests. The other tests were integration tests (they use simapp) and are at `tests/integration/staking/keeper/unbonding_test.go`